### PR TITLE
refactor sprint 8

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR refactors
- [x] replace from_str by from
- [x] Fix warnings in crate
- [x] spelleing: occurence -> occurrence
- [x] &*[a-z,A-Z].modulus.modulus -> [a-z,A-Z].modulus.get_fmpz_mod_ctx_struct()
- [x] warning for CMP of PolyOverZq
- [x] Modulus Error in set entry of MatZq in missing (maybe in other modules too)
- [x] same as in get module, description in set module must be changed
- [x] MatPolyOverZ does not use macros (for example: set)
- [x] close TODO in new of matrices (use indices not index)
- [x] Add `Return ...` for functions returning `Result<(), MathError>`

**Testing**

- [ ] I added basic working examples (possibly in doc-comment)
- [ ] I added tests for large (pointer representation) values
- [ ] I triggered all possible errors in my test in every possible way
- [ ] I included tests for all reasonable edge cases

**Checklist:**

- [ ] I have performed a self-review of my own code
  - [ ] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [ ] The chosen implementation is not more complex than it has to be
  - [ ] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [ ] The doc comments fit our style guide
  - [ ] I have credited related sources if needed

### CODE REVIEW

File: src/utils/index.rs [hunk 1]
```diff
@@ -69,6 +69,36 @@ pub fn evaluate_index<S: TryInto<i64> + Display>(index: S) -> Result<i64, MathEr
...
+pub fn evaluate_indices(
+    index1: impl TryInto<i64> + Display,
+    index2: impl TryInto<i64> + Display,
+) -> Result<(i64, i64), MathError> {
+    let row_i64 = evaluate_index(index1)?;
+    let column_i64 = evaluate_index(index2)?;
+
+    Ok((row_i64, column_i64))
+}
+
```
Comment (3pmile) [inline src/utils/index.rs:101]: is this still in for a reason? if yes, why no tests

**Review by 3pmile — CHANGES_REQUESTED** [review]

File: src/utils/index.rs [hunk 2]
```diff
@@ -69,6 +69,36 @@ pub fn evaluate_index<S: TryInto<i64> + Display>(index: S) -> Result<i64, MathEr
...
+///
+/// # Errors and Failures
+/// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds) if
+/// either the index is negative or it does not fit into an [`i64`].
+pub fn evaluate_indices(
+    index1: impl TryInto<i64> + Display,
+    index2: impl TryInto<i64> + Display,
+) -> Result<(i64, i64), MathError> {
+    let row_i64 = evaluate_index(index1)?;
+    let column_i64 = evaluate_index(index2)?;
```
Comment (3pmile) [inline src/utils/index.rs:97]: naming?

File: src/integer/mat_z/sample/uniform.rs [hunk 3]
```diff
@@ -134,7 +133,7 @@ mod test_sample_uniform {
     /// implementing Into<Z> + Clone, i.e. u8, u16, u32, u64, i8, ...
     #[test]
     fn availability() {
-        let modulus = Modulus::from_str("7").unwrap();
+        let modulus = Modulus::try_from_z(&Z::from(7)).unwrap();
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/mat_z/sample/uniform.rs:136]: Yes, that`s why I wrote it in the next refactoring

File: src/integer/z/cmp.rs [hunk 4]
```diff
@@ -162,8 +162,8 @@ mod test_partial_eq {
         assert!(small_1 == small_2);
         assert!(small_2 == small_1);
         assert!(small_1 == small_1);
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/z/cmp.rs:164]: Why? They are equal, but clippy wants us not to use unnecessary brackets.

File: src/rational/q/cmp.rs [hunk 5]
```diff
@@ -225,29 +225,29 @@ mod test_partial_eq {
...
-        let small_positive = Q::from_str("1").unwrap();
-        let small_negative = Q::from_str("-1").unwrap();
+        let small_positive = Q::ONE;
+        let small_negative = Q::MINUS_ONE;
         let large_negative_str = format!("-{:1<65}", "1");
         let min = Q::from_str(&large_negative_str).unwrap();
 
-        assert!(!(max == small_negative));
-        assert!(!(small_negative == max));
-        assert!(!(max == small_positive));
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/rational/q/cmp.rs:235]: s.o.

File: src/integer/z/cmp.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (3pmile) [reply src/integer/z/cmp.rs:164]: Then please move the tests to the right test case or remove them entirely

[the remark this answers is not in the document]
Comment (3pmile) [reply src/integer/z/cmp.rs:164]: Equal tests are now in the not_equal_... test case and the other way around

**Review by 3pmile — APPROVED** [review]

**Pull request merged**